core-bundle/tests/Command/UserListCommandTest.php

```diff
@@ -106,62 +105,21 @@ public function testTakesColumnAsArgument(): void
 
     private function getCommand(): UserListCommand
     {
-        $collection = new Collection([$this->mockAdminUser(), $this->mockContaoUser()], 'tl_user');
-
-        $userModelAdapter = $this->mockAdapter(['findBy', 'findAll']);
-        $userModelAdapter
-            ->method('findAll')
-            ->willReturn($collection, null)
+        $qb = $this->createMock(QueryBuilder::class);
+        $qb
+            ->method('select')
+            ->willReturnSelf()
         ;
 
-        $collection = new Collection([$this->mockAdminUser()], 'tl_user');
-
-        $userModelAdapter
-            ->method('findBy')
-            ->with('admin', '1')
-            ->willReturn($collection, null)
+        $connection = $this->createMock(Connection::class);
+        $connection
+            ->method('createQueryBuilder')
+            ->willReturn($qb)
         ;
 
-        $command = new UserListCommand($this->mockContaoFramework([UserModel::class => $userModelAdapter]));
+        $command = new UserListCommand($connection);
         $command->setApplication(new Application());
 
         return $command;
     }
-
-    /**
-     * @return UserModel&MockObject
-     */
-    private function mockContaoUser(): UserModel
-    {
-        $userModel = $this->mockClassWithProperties(UserModel::class);
-        $userModel->id = 2;
-        $userModel->username = 'j.doe';
-        $userModel->name = 'John Doe';
-
-        $userModel
-            ->method('row')
-            ->willReturn((array) $userModel)
-        ;
-
-        return $userModel;
-    }
-
-    /**
-     * @return UserModel&MockObject
-     */
-    private function mockAdminUser(): UserModel
-    {
-        $userModel = $this->mockClassWithProperties(UserModel::class);
-        $userModel->id = 1;
-        $userModel->username = 'j.doe';
-        $userModel->name = 'John Doe';
-        $userModel->admin = '1';
-
-        $userModel
-            ->method('row')
-            ->willReturn((array) $userModel)
-        ;
-
-        return $userModel;
-    }
```

Comment:
We should probably mock `$qb->fetchAllAssociative()` to return these users to test the command properly.

Reply:
they have not been tested at all yet either 😂 🤷

Reply:
So what? This is never an argument - we can always improve things.

Reply:
> they have not been tested at all yet either 😂 🤷

But at least the two `foreach ($users as $user)` statements got executed in the test ☺️
But I agree we should add simple calls like `->assertContains('j.doe', $output)` for the text output and verify the contents directly for the JSON output.

Reply:
I've rewritten all tests since the current ones were pretty useless.